pyat/at/lattice/particle_object.py

```diff
@@ -75,3 +92,59 @@ def rest_energy(self) -> numpy.ndarray:
     def charge(self) -> float:
         """Particle charge [elementary charge]"""
         return self._charge
+
+    @property
+    def harmonic_number(self):
+        return self._harmn
+
+    @harmonic_number.setter
+    def harmonic_number(self, harmn):
+        self._harmn = harmn
+        self.fillpattern = 1
+
+    @property
+    def fillpattern(self):
+        return self._fillpattern
+
+    @fillpattern.setter
+    def fillpattern(self, bunches):
+        if (self._harmn is None):
+            warn(UserWarning("Harmonic number not set in Beam(), "
+                             "nbunch and fillpattern set to 1"))
+            fp = numpy.ones(1)
+        elif numpy.isscalar(bunches):
+            if bunches == 1:
+                fp = numpy.ones(1)
```

Comment:
the output of np.ones is a float (1.0). Is it not more appropriate for filling pattern to be a mask? In which case you could use np.ones(992, dtype=bool). It is still compatible when multiplied with weights as True*2 = 2

Reply:
`Particle.fillingpattern` now return a mask